java-manta-client-unshaded/src/main/java/com/joyent/manta/client/PruneEmptyParentDirectoryStrategy.java

```diff
@@ -37,34 +35,68 @@ private PruneEmptyParentDirectoryStrategy() {
     }
 
     /**
-     * This will delete empty parent directories up to the root.
+     * Method that will delete the specified path. If the delete of the path is
+     * successful, then it will progressively go from right (deepest) to left
+     * in the path attempting to delete all parent directories until a non-empty
+     * directory is encountered or until a directory is missing.
      *
-     * @param client - A valid client.
-     * @param path - THe complete path that you want to start with.
-     * @param limit - The limit of directories to delete.
-     * @throws IOException - this is thrown from
+     * @param client a reference to the manta client instance for the operations
+     * @param path file or directory object to delete
+     * @param limit limit of directories to delete
```

Comment:
Missing @param headers

Reply:
I meant in the javadoc comment for the method, there are 4 params, but the comment lists only 3 and exception. Not a biggie.